src/Workflow/ItemInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+
+namespace PHPMentors\Workflower\Workflow;
+
+
+use PHPMentors\Workflower\Workflow\Activity\ActivityInterface;
+
+interface ItemInterface
+{
+    /**
+     * @return int|string
+     */
+    public function getId();
+
+    /**
+     * @param ProcessInstanceInterface $processInstance
+     * @return void
+     */
+    public function setParentProcessInstance(ProcessInstanceInterface $processInstance);
+
+    /**
+     * @return ProcessInstanceInterface|null
+     */
+    public function getParentProcessInstance();
```

Comment:
It might be `setEnclosingProcessInstance/getEnclosingProcessInstance`, but in this case `setProcessInstance/getProcessInstance` is better.

Reply:
I used `parentProcessInstance` to make it more obvious that a sub-process instance can have a parent process instance. Same logic for `parentActivity`.

Reply:
Let's use a prefix. The prefix could be `parent`, `owner` or `enclosing` but we definitely should have a prefix to avoid confusion for both `processInstance` and `activity`.

Reply:
On a 2nd thought I think that we might just go with `setProcessInstance/getProcessInstance` and `setActivity/getActivity`. It sounds better indeed and if we document the functions better then it's all good.